src/Package/Impl/DataInspect/Viewers/FunctionViewer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.Common.Core;
+using Microsoft.R.Core.Formatting;
+using Microsoft.R.Debugger;
+using Microsoft.R.Editor.Settings;
+using Microsoft.R.Editor.Signatures;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.R.Package.Repl;
+using Microsoft.VisualStudio.R.Package.Shell;
+using static System.FormattableString;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect.Viewers {
+    [Export(typeof(IObjectDetailsViewer))]
+    internal sealed class FunctionViewer : IObjectDetailsViewer {
+        private readonly IRSessionProvider _sessionProvider;
+        private readonly IDebugObjectEvaluator _evaluator;
+
+        [ImportingConstructor]
+        public FunctionViewer(IRSessionProvider sessionProvider, IDebugObjectEvaluator evaluator) {
+            _sessionProvider = sessionProvider;
+            _evaluator = evaluator;
+        }
+
+        #region IObjectDetailsViewer
+        public ViewerCapabilities Capabilities => ViewerCapabilities.Function;
+
+        public bool CanView(DebugValueEvaluationResult evaluation) {
+            return evaluation != null && evaluation.Classes.Count == 1 && evaluation.Classes[0].EqualsOrdinal("function");
+        }
+
+        public async Task ViewAsync(string expression, string title) {
+            var evaluation = await _evaluator.EvaluateAsync(expression, DebugEvaluationResultFields.Expression) as DebugValueEvaluationResult;
+            if(evaluation == null) {
+                return;
+            }
+
+            string functionName = evaluation.Expression as string;
+            if (string.IsNullOrEmpty(functionName)) {
+                return;
+            }
+
+            var session = _sessionProvider.GetInteractiveWindowRSession();
+            string functionCode = null;
+            using (var e = await session.BeginEvaluationAsync()) {
+                var result = await e.EvaluateAsync(Invariant($"paste0(deparse({functionName}), collapse='\n')"), REvaluationKind.Normal);
```

Comment:
Use `EvaluateAsync<string>()` - you get correct Unicode handling that way, and errors will be thrown as `RException`.